pkg/builder/AppDetailBuilder.go

```diff
@@ -407,6 +404,41 @@ func UpgradeRelease(request *client.UpgradeReleaseRequest) (*client.UpgradeRelea
 	return upgradeReleaseResponse, nil
 }
 
+func GetDeploymentDetail(request *client.DeploymentDetailRequest) (*client.DeploymentDetailResponse, error) {
+	conf, err := k8sUtils.GetRestConfig(request.ReleaseIdentifier.ClusterConfig)
+	if err != nil {
+		return nil, err
+	}
+	opt := &helmClient.RestConfClientOptions{
+		Options: &helmClient.Options{
+			Namespace: request.ReleaseIdentifier.ReleaseNamespace,
+		},
+		RestConfig: conf,
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	helmClient, err := helmClient.NewClientFromRestConf(opt)
+	if err != nil {
+		return nil, err
+	}
+
+	releases, err := helmClient.ListReleaseHistory(request.ReleaseIdentifier.ReleaseName, 20)
+	if err != nil {
+		return nil, err
+	}
+	manifest := ""
+	for _, helmRelease := range releases {
+		if request.DeploymentVersion == int32(helmRelease.Version) {
+			manifest = helmRelease.Manifest
```

Comment:
break statement and set valuesYaml is missing. 
(TIP : you can extract some part from GetHelmAppValues method and use that)